### PR TITLE
Add auto-generated docs for custom math examples

### PR DESCRIPTION
**Pull request by sjpfenninger**
Summary of changes in this pull request:

* Changed comment strings into commented-out YAML with a "title" and "description" for each example (the `# ---` comment marks the end of the "docs comment")
* Added code to auto-generate custom math documentation

Reviewer checklist:

- [ ] Test(s) added to cover contribution
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Coverage maintained or improved

### CODE REVIEW

File: doc/helpers/generate_custom_constraints.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    for yaml_file in sorted(custom_math_dir.glob("*.yaml")):
+        rst_string = rst_string + process_file(yaml_file)
+
+    with open("_generated/custom_math_examples.rst", "w") as f:
+        f.write(rst_string)
+
+
+def process_file(path):
+    with open(path, "r") as f:
```
Comment (brynpickering) [inline doc/helpers/generate_custom_constraints.py:60]: `path.read_text()` handles this without the need for the context manager.

File: doc/helpers/generate_custom_constraints.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Copyright (C) since 2013 Calliope contributors listed in AUTHORS.
+# Licensed under the Apache 2.0 License (see LICENSE file).
+
+"""
+generate_custom_constraints.py
```
Comment (brynpickering) [inline doc/helpers/generate_custom_constraints.py:5]: `generate_custom_math_examples`?

File: doc/helpers/generate_custom_constraints.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generates an `.rst` file listing all custom constraints.
```
Comment (brynpickering) [inline doc/helpers/generate_custom_constraints.py:8]:
```suggestion
Generates an `.rst` file listing all custom math examples.
```

File: doc/_static/custom_math/uptime_downtime_limits.yaml [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
-# The `downtime_period` constraint can be used if you know a specific period in time where your technology will need to be down for maintenance and can be defined by setting a value in those timesteps, leaving all other timesteps empty (i.e., NaN).
-# The `downtime_period_decision` constraint will allow a technology with integer decision variables activated to decide on a certain number of timesteps in the year to not be operating, although it is not possible to enforce that these downtime periods are consecutive.
+# title: Up-/downtime limits
 
-# New technology-level parameters
-# capacity_factor_min
-# capacity_factor_max
-# downtime_periods (from CSV as a timeseries)
-# uptime_limit
+# description: |
```
Comment (brynpickering) [inline doc/_static/custom_math/uptime_downtime_limits.yaml:3]: the pipe leave the result with the line breaks when loaded in, `>` ignores the line breaks. Do we have a preference?

**Review by brynpickering — CHANGES_REQUESTED** [review]

File: doc/_static/custom_math/uptime_downtime_limits.yaml (hunk 4 above)
Comment (sjpfenninger) [reply doc/_static/custom_math/uptime_downtime_limits.yaml:3]: The pipe is needed to keep formatting like paragraphs intact, so I think we should always go with the pipe.

**Comment by sjpfenninger** [comment]
Closing this because it is superseded by #526 (which addresses all of the issues pointed out in the review, with the possible exception of the question on `|` vs `>`)

**Pull request closed without merging**